src/examples/example_event_error_details.rs

```diff
@@ -0,0 +1,100 @@
+use std::sync::mpsc::channel;
+
+use clap::{load_yaml, App};
+use codec::Decode;
+use keyring::AccountKeyring;
+use sp_core::crypto::Pair;
+use sp_runtime::AccountId32 as AccountId;
+use sp_runtime::MultiAddress;
+
+use substrate_api_client::rpc::WsRpcClient;
+use substrate_api_client::sp_runtime::app_crypto::sp_core::sr25519;
+use substrate_api_client::{Api, ApiResult, XtStatus};
+
+// Look at the how the transfer event looks like in in the metadata
+#[derive(Decode)]
+struct TransferEventArgs {
+    from: AccountId,
+    to: AccountId,
+    value: u128,
+}
+
+fn main() {
+    env_logger::init();
+    let url = get_node_url_from_cli();
+
+    // initialize api and set the signer (sender) that is used to sign the extrinsics
+    let from = AccountKeyring::Alice.pair();
+
+    let client = WsRpcClient::new(&url);
+    let api = Api::<sr25519::Pair, _>::new(client)
+        .map(|api| api.set_signer(from.clone()))
+        .unwrap();
+
+    let from_account_id = AccountKeyring::Alice.to_account_id();
+    match api.get_account_data(&from_account_id).unwrap() {
+        Some(alice) => println!("[+] Alice's Free Balance is is {}\n", alice.free),
+        None => println!("[+] Alice's Free Balance is is 0\n"),
+    }
+
+    let to = AccountKeyring::Bob.to_account_id();
+
+    match api.get_account_data(&to).unwrap() {
+        Some(bob) => println!("[+] Bob's Free Balance is is {}\n", bob.free),
+        None => println!("[+] Bob's Free Balance is is 0\n"),
+    }
+    // generate extrinsic
+    let amount = 1000000000000000000;
+    let xt = api.balance_transfer(MultiAddress::Id(to.clone()), amount);
+
+    println!(
+        "Sending an extrinsic from Alice (Key = {}),\n\nto Bob (Key = {})\n",
+        from.public(),
+        to
+    );
+    println!("[+] Composed extrinsic: {:?}\n", xt);
+
+    // send and watch extrinsic until finalized
+    let tx_hash = api
+        .send_extrinsic(xt.hex_encode(), XtStatus::InBlock)
+        .unwrap();
+    println!("[+] Transaction got included. Hash: {:?}\n", tx_hash);
+
+    let (events_in, events_out) = channel();
+    api.subscribe_events(events_in).unwrap();
+    let args: ApiResult<TransferEventArgs> =
+        api.wait_for_event("Balances", "Transfer", None, &events_out);
+    match args {
```

Comment:
Jenkins Output:
```log
+ target/release/examples//example_event_error_details
Interacting with node on ws://127.0.0.1:9944

[+] Alice's Free Balance is is 1152921453830787971

[+] Bob's Free Balance is is 1152921505007848018

Sending an extrinsic from Alice (Key = 5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY),

to Bob (Key = 5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty)

[+] Composed extrinsic: UncheckedExtrinsic(Some((MultiAddress::Id(d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d (5GrwvaEF...)), GenericExtra(Era::Immortal, 403, 0))), ([5, 0], MultiAddress::Id(8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48 (5FHneW46...)), 1000000000000000000))

[+] Transaction got included. Hash: Some(0x07fa345ac583560385e3861dd82c41f785807ba577ac3bb88bd5a7fa8c8e21b9)

Error : Transfer event received!!!

Transactor: d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d (5GrwvaEF...)
Destination: 8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48 (5FHneW46...)
Value: 1000000000000000000
[+] Bob's Free Balance is now 2152921505105848018

[+] Alice's Free Balance is now 152921441357773710
```

is that expected?

Reply:
@clangenb I'd love to hear your opinion on this matter:

Jenkins uses the examples as tests against the node, however, most examples only print the output but don't actually test anything else than connecting to the node and receivin the events. So upon unexpected events (such as in this example ; the transfer event was successful) it fails to fail.

Are the examples here for the examples sake or should we modify them to actually test something? (ofc in a different issue, not in this one)

Reply:
Sorry. The amount to transfer is not big enough, so the transfer coud be executed. I will increase the value.

Reply:
The primary purpose of the example is to verify that we can successfully talk to the node, and also to show how things are done. Implicitly, we have some testing because we `unwrap` many times. I have checked the other examples, and I believe that unwraps suffice.

Exact testing would require some effort, as the return value would depend on the state of the blockchain, which means we can only meaningfully test on a clean node, so you can't run the example twice on the same node. Additionally, some tests currently depend on each other, e.g,  Alice's nonce is continuously increasing.